tools/upload_text.php

```diff
@@ -343,6 +344,26 @@ function process_file($project, $indicator, $stage, $returning_to_pool)
                     throw new FileUploadException("failed to extract files");
                 }
             }
+            // if there is an htm or html file, zip it with images directory
+            $htm_files = glob("$smooth_dir/*.{htm,html}", GLOB_BRACE);
+            if($htm_files)
+            {
+                $htm_file = $htm_files[0];
```

Comment:
What happens if there is more than one file? This would take the first one which could be a random one. Should we put all of them in? Throw an error if there is more than one?

Reply:
I would say that if there is more than one HTML file, zip them all into the `HTML+images` zip for download. The problem is that PPers still haven't gotten the hang of zipping the HTML and images together for inclusion into the master SR upload zip. For PPV, they don't have to, and for PG they don't have to. This would bring the SR upload more in line with what PPV and PG accept.

It's not going to be perfect, whatever is done, but at least this will align PPers' expectations more with what they're used to.

Reply:
If they had names like "book.html" and "book_no_images.html" then the smooth reader would know what they were for when the file was unzipped. But what would we name the zip file? Perhaps <projectID>.zip.

Reply:
Hm. Good point. Do we keep a copy of what they named the zip to begin with? If we do, we could use that, with `-html` appended to the filename, preceding the extension. That would be my first choice. IIf that's too annoying to code, then `projectID-html.zip` would be my choice.

Reply:
We do know the original name. I'll try what you suggest.

Reply:
This change has now been made.